### PR TITLE
Add snap cards on bottom of the map with details about displayed hotels

### PR DESCRIPTION
**Pull request by jaroslav-kubicek**
screen shot 2017-12-18 at 14 18 41

*Note*  - there is one todo related to flow types to be done

Any feedback from user testing on how it feels when you are using it, is welcomed 🙂

### CODE REVIEW

**Comment by mrtnzlml** [comment]
Please test these changes in [Expo](https://docs.expo.io/versions/latest/introduction/installation.html#mobile-client-expo-for-ios-and-android) after build has finished:

  QR Code

**Comment by jaroslav-kubicek** [comment]
omg, FYI device orientation needs to be fixed...

**Comment by mrtnzlml** [comment]
I don't know man. It feels so *not smooth*. I always drag like I would normally do but it waits for a while and that it finishes the movement with higher speed. Feels so weird.

Also, at some point, it scrolls to the very beginning (very fast). But if you continue long enough with scrolling it will never happen again (not nice).

Is the address not changing at all or is it still the same so I cannot see the effect?

> Device orientation needs to be fixed

Please do so.

File: app/common/src/map/Address.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class Address extends React.Component<Props> {
+  render = () => {
+    return (
+      <View style={styles.container}>
+        <View style={styles.mapIcon}>
+          <Ionicons name="md-map" size={24} color={Color.brand} />
+        </View>
+        <View>
+          <Text style={{ fontWeight: 'bold' }}>Address</Text>
+          <Text>950 Candice Wells Suite 375, Dublin</Text>
```
Comment (mrtnzlml) [inline app/common/src/map/Address.js]: I would expect props here instead of text.

File: app/common/src/map/HotelSnap.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    color: Color.grey._700,
+    opacity: 0.8,
+  },
+  price: {
+    color: Color.brand,
+    fontWeight: 'bold',
+  },
+});
+
+const HotelSnap = ({ width, hotel }: Props) => (
```
Comment (mrtnzlml) [inline app/common/src/map/HotelSnap.js]: Can we find a better name for this? Other than "snap".

File: app/hotels/src/map/HotelSnaps.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  componentDidMount = () => {
+    Dimensions.addEventListener('change', this.onDimensionsChanged);
+  };
+
+  componentWillUnmount = () => {
+    Dimensions.removeEventListener('change', this.onDimensionsChanged);
+  };
+
+  onDimensionsChanged = ({ screen: { width } }: OnDimensionsChange) => {
+    this.setState({ availableWidth: width });
```
Comment (mrtnzlml) [inline app/hotels/src/map/HotelSnaps.js]: Why is this needed? Cannot we just use "width" from `Dimensions`? The trick is that you have to use it inside of render function so it gets updated every time you rerender (this is what you want anyway).

File: app/hotels/src/map/HotelSnaps.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    const { availableWidth } = this.state;
+    const width = availableWidth * SNAP_WIDTH;
+
+    return <HotelSnap width={width} hotel={item.data} />;
+  };
+
+  render = () => {
+    const { items, selectedIndex, onSnapToItem } = this.props;
+    const { availableWidth } = this.state;
+    const itemWidth = availableWidth * SNAP_WIDTH;
```
Comment (mrtnzlml) [inline app/hotels/src/map/HotelSnaps.js]: Seems like this is calculated twice (render and renderItem). Try to be DRY... 🙂

File: app/hotels/src/map/HotelSnaps.js (hunk 3 above)
Comment (jaroslav-kubicek) [reply app/hotels/src/map/HotelSnaps.js]: IMHO component is not re-rendered because the device has rotated, I even tried that, that's why I added lisneter, but I may miss something you already know...

Comment (mrtnzlml) [reply app/hotels/src/map/HotelSnaps.js]: I did it here: https://github.com/kiwicom/react-native-app/blob/master/app/hotels/src/gallery/PhotosStripe.js#L23 (just to fix exactly the same issue)

Comment (mrtnzlml) [reply app/hotels/src/map/HotelSnaps.js]: It's described here: https://facebook.github.io/react-native/docs/dimensions.html#get But now I really don't know why does it work... ❓

File: app/common/src/map/Address.js (hunk 1 above)
Comment (jaroslav-kubicek) [reply app/common/src/map/Address.js]: fixed

File: app/hotels/src/map/HotelSnaps.js (hunk 4 above)
Comment (jaroslav-kubicek) [reply app/hotels/src/map/HotelSnaps.js]: fixed

File: app/hotels/src/map/HotelSnaps.js (hunk 3 above)
Comment (mrtnzlml) [reply app/hotels/src/map/HotelSnaps.js]: Ok I bet it's about `onLayout` change in the `Swiper` so it doesn't work exactly how I thought... 🙂

File: app/common/src/map/HotelSnap.js (hunk 2 above)
Comment (jaroslav-kubicek) [reply app/common/src/map/HotelSnap.js]: fixed (hopefully)

File: app/common/src/map/HotelCard.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+});
+
+const HotelCard = ({ width, hotel }: Props) => (
+  <View style={[{ width }]}>
+    <View style={styles.description}>
+      <View>
+        <Image
+          style={styles.image}
+          source={{
+            uri: 'http://q.bstatic.com/images/hotel/square128/117/11753593.jpg',
```
Comment (mrtnzlml) [inline app/common/src/map/HotelCard.js]: Should I care now or we will connect the real data later?

Comment (jaroslav-kubicek) [reply app/common/src/map/HotelCard.js]: I plan to do new PR for real data :)

**Comment by jaroslav-kubicek** [comment]
> I don't know man. It doesn't feel very smooth. I always drag like I would normally do but it waits for a while and that it finishes the movement with higher speed. Feels so weird.

I removed momentum prop and added `decelerationRate`, which seems to be slightly smoother

> Also, at some point, it scrolls to the very beginning

valid point when `firstItem !== 0`, I expect `firstItem === 0` by default when you simply navigate to map by hitting "map" button in topbar, will investigate to make it work always.

Add resizing - I tried and it doesn't work without listeners :/

**Comment by jaroslav-kubicek** [comment]
Related issue: https://github.com/archriss/react-native-snap-carousel/issues/232

**Comment by mrtnzlml** [comment]
Let me know about the progress. It's not OK to merge obviously broken code...

**Comment by jaroslav-kubicek** [comment]
workarounded :) cc @mrtnzlml

File: app/common/index.js [hunk 6]
```diff
@@ -20,16 +20,26 @@ export { default as NetworkImage } from './src/image/NetworkImage';
 export { default as FullPageLoading } from './src/loaders/FullPageLoading';
 
 export { default as PriceMarker } from './src/map/PriceMarker';
+export { default as HotelCard } from './src/map/HotelCard';
+export { default as Address } from './src/map/Address';
```
Comment (mrtnzlml) [inline app/common/index.js]: I do not think this should be in this shared package "just in case". I do not think there is high chance that we'll reuse these components. It's absolutely fine to create local components and use this only for true "shared" things. I am actually considering rename `common` to `shared` (like in React). I'll leave it here for now. I just want to mention it... 🙂

**Comment by mrtnzlml** [comment]
> workarounded

I can reproduce it again just with a different flow. Try to click few times on different map markers and then try to swipe. It's broken again. I'll merge it for now because everyone who is not actively creating this project is bitching around but we will need to spend few more hours on this swiper. It's not very good yet.

**Comment by mrtnzlml** [comment]
There is also this warning:

> VirtualizedList: You have a large list that is slow to update - make sure your renderItem function renders components that follow React performance best practices like PureComponent, shouldComponentUpdate, etc.
Object {
   "contentLength": 6003,
   "dt": 2661,
   "prevDt": 1851,
 }

**Comment by mrtnzlml** [comment]
Merged.

File: app/common/index.js (hunk 6 above)
Comment (jaroslav-kubicek) [reply app/common/index.js]: good point, I'm actually going to change it today :)
